roles/openshift_on_openstack/tasks/main.yml

```diff
@@ -188,50 +201,84 @@
     dest: "{{ osev3_yml_path }}"
     backup: yes
 
-
-# Get some debug information that makes it easier to file bugs.
-- name: Running commands that create debug information needed to open issues
-  shell: "{{ item }}"
-  register: debug_information
-  with_items:
-    - "printf \"$(uname -srm)\n$(cat /etc/os-release)\n\""
-    - "ansible --version"
-    - "cd openshift-ansible ; git describe"
-    - "rpm -q atomic-openshift-utils openshift-ansible"
-  changed_when: false
+- name: Copy {{ osev3_yml_path }} to the artifacts directory
+  synchronize:
+    src: "{{ osev3_yml_path }}"
+    dest: "{{ artifacts_directory }}"
+    mode: pull
   ignore_errors: true
 
-# Print the debug information out.
-- name: Printing the debug information to the screen
-  debug:
-    msg: "{{ item['stdout_lines'] }}"
-  with_items: "{{ debug_information['results'] }}"
-
 # Set the path for the prerequisite log file.
 - name: Creating a variable for the prerequisite log file
   set_fact:
     prerequisite_log: "{{ ansible_user_dir }}/openshift_prerequisites.log"
 
-# Run the Ansible playbook that checks the prerequisites in OpenStack.
-- name: Checking the prerequisites for cluster provisioning in OpenStack
-  # Using two inventory flags here as the README instructs.
-  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/prerequisites.yml 2>&1 >> {{ prerequisite_log }}"
-  args:
-    # Use bash to get the posix style redirects.
-    executable: /bin/bash
+- name: OpenShift prerequisite block
+  block:
+    # Run the Ansible playbook that checks the prerequisites in OpenStack.
+    - name: Checking the prerequisites for cluster provisioning in OpenStack
+      # Using two inventory flags here as the README instructs.
+      shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/prerequisites.yml 2>&1 >> {{ prerequisite_log }}"
+      args:
+        # Use bash to get the posix style redirects.
+        executable: /bin/bash
+  rescue:
```

Comment:
For clarity, the rescue statement "undoes" the if the playbook above fails.  We can use the rescue statement to run some debugging commands afterwards as you have below, but I think the intent here would want to stick an explicit fail at the end of the rescue portion of this block.  This would still fail the playbook.  (The always portion of the block will always run, even if you explicitly fail inside the rescue portion.)

Reply:
My testing showed that if the task in the block failed the rescue code ran and the playbook still failed. I can check it again.

Reply:
Can you attach the the failed output or send it to me in a dm.  I would suspect that the failure is related to the above double indentation (If that wasn't caught before running the playbook) or there is an error inside the rescue portion.